### PR TITLE
Handle deprecations for 5.0

### PR DESCRIPTION
**Pull request by gasman**
Handle most of the deprecations in 5.0. The one for StreamField `use_json_field` needs more thought, so doing that one separately.

NB Not to be merged until https://github.com/wagtail/bakerydemo/pull/386 is merged, as removing wagtail.core will break bakerydemo otherwise (wagtailfontawesome contains wagtail.core imports).

### CODE REVIEW

**Comment by lb-** [comment]
I have merged in https://github.com/wagtail/bakerydemo/pull/386

File: wagtail/models/__init__.py [hunk 1]
```diff
@@ -2896,15 +2862,6 @@ def save(self, user=None, *args, **kwargs):
     def as_object(self):
         return self.content_object.with_content_json(self.content)
 
-    def as_page_object(self):
```
Comment (laymonage) [inline wagtail/models/__init__.py:2899]: Shall we remove the admonition from the docs as well?

https://github.com/wagtail/wagtail/blob/b336536015da7c3ecf1c5249e4fb3cc8062ca9d7/docs/reference/pages/model_reference.md?plain=1#L801-L803

Not sure how long we should keep these, but we also still have 3.0 stuff as well that should probably be removed.

**Review by laymonage — COMMENTED** [review]
These changes look good to me, but I'm wondering whether we should also remove old `versionchanged` notes from the docs, especially those from 3.0 and 4.0.

Oh, and should we also explicitly list all these removals in the upgrade considerations, or do you think raising the warning was enough to signify things that needed to be removed?

**Comment by gasman** [comment]
@laymonage Have now removed the old `versionadded` / `versionchanged` annotations and added release notes. (Was meaning to do the version annotations as the last step, but that got forgotten when I broke off the `use_json_field` change into its own PR, oops...)

File: docs/reference/management_commands.md [hunk 2]
```diff
@@ -6,11 +6,6 @@
 
 ## publish_scheduled
 
-```{versionchanged} 4.1
```
Comment (laymonage) [inline docs/reference/management_commands.md:9]: Are we keeping this alias indefinitely, or should we start a deprecation process?

**Review by laymonage — APPROVED** [review]
This looks good to me, just curious on what we should do with the `publish_scheduled_pages` command. If we deprecate it, we can do it in a separate PR.

**Pull request merged**